core/internal/runhistory/runhistory.go

```diff
@@ -0,0 +1,48 @@
+package runhistory
+
+import (
+	"fmt"
+
+	json "github.com/wandb/simplejsonext"
+	"github.com/wandb/wandb/core/internal/pathtree"
+	"github.com/wandb/wandb/core/pkg/service"
+)
+
+type RunHistory struct {
+	*pathtree.PathTree[*service.HistoryItem]
+	Step *int64
+}
+
+func New(step *int64) *RunHistory {
+	return &RunHistory{
+		PathTree: pathtree.New[*service.HistoryItem](),
+		Step:     step,
+	}
+}
+
+// Serializes the object to send to the backend.
+func (runHistory *RunHistory) Serialize(format pathtree.Format) ([]byte, error) {
+	// A configuration dict in the format expected by the backend.
+	return runHistory.PathTree.Serialize(format, func(value any) any {
+		return value
+	})
+}
+
+func (runHistory *RunHistory) Flatten() ([]*service.HistoryItem, error) {
+	var items []*service.HistoryItem
+	for _, item := range runHistory.PathTree.Flatten() {
+		value, err := json.Marshal(item.Value)
```

Comment:
gotta test with all kinds of weird values like nans or infs etc

Reply:
we do here: https://github.com/wandb/wandb/blob/237aed85c1cfcc2c28f268e721179976e058e063/tests/pytest_tests/system_tests/test_core/test_wandb_run.py#L14

but i guess i can add some to the handler_test as well